src/pages/dashboard.tsx

```diff
@@ -0,0 +1,129 @@
+  const parsedStatuses = {
+    IN_PROGRESS: "Continue Your Application",
+    PENDING_REVIEW: "View Application",
+    IN_REVIEW: "View Application",
+    ACCEPTED: "RSVP by XXXX",
+    REJECTED: "",
+    WAITLISTED: "View Application",
+    DECLINED: "Change RSVP",
+  };
+
+  return parsedStatuses?.[status ?? "IN_PROGRESS"];
+};
+
+const Dashboard = () => {
+  const { data: application } = api.application.get.useQuery();
+  const router = useRouter();
+
+  const onButtonClick = (status: ApplicationStatusType) => {
+    switch (status) {
+      case "IN_PROGRESS":
+        void router.push("/apply?step=persona");
+      case "PENDING_REVIEW":
+        void router.push("/apply?step=review");
+      default:
+        void router.push("/apply?step=review");
+    }
+  };
+
+  return (
+    <>
+      <Head>
+        <title>Hack Western</title>
+        <meta
+          name="description"
+          content="Hack Western: One of Canada's largest annual student-run hackathons based out of Western University in London, Ontario."
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <main className="flex h-svh max-h-svh flex-col items-center bg-primary-50">
+        <ApplyNavbar />
+        <div className="flex w-full flex-1 flex-grow items-center">
+          <div
+            id="left-panel"
+            className="lg:w-xl flex h-full flex-grow flex-col justify-center gap-4 bg-primary-100 p-9 pt-12 text-center lg:max-w-xl"
+          >
+            <div className="pb-2.5 text-3xl font-bold text-slate-700">
+              {
+                // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                `Welcome Back, ${application?.firstName || "Hacker"}!`
+              }
+            </div>
+            <div className="text-xl font-medium text-slate-700">Status:</div>
+            <ApplicationStatus status={application?.status ?? "IN_PROGRESS"} />
+            {application?.status !== "REJECTED" ? (
+              <div
+                className="mx-auto mt-6 w-max cursor-pointer select-none rounded-2xl bg-primary-600 px-4 py-2.5 text-lg text-white transition-all hover:bg-primary-500"
+                onClick={() =>
+                  onButtonClick(application?.status ?? "IN_PROGRESS")
+                }
+              >
+                {buttonStatus(application?.status ?? "IN_PROGRESS")}
+              </div>
+            ) : (
+              <div className="text-slate-700">
+                Due to the volume of applicants, we were unable to accept
+                everyone. That doesn&apos;t mean it&apos;s the end of your Hack
+                Western journey. We encourage you to apply again next year!
+              </div>
+            )}
+          </div>
+          <div
+            id="right-panel"
+            className="flex h-full w-[60vw] flex-col items-center justify-center"
+          >
+            <Passport />
+          </div>
+        </div>
+      </main>
+    </>
+  );
+};
+
```

Comment:
we should add the authRedirect getServerSiderProps to this page

Reply:
facts